### PR TITLE
[llvm] Use llvm::sort (NFC)

### PR DESCRIPTION
**Pull request by kazutakahirata**

### CODE REVIEW

File: llvm/lib/Target/AMDGPU/AMDGPUIGroupLP.cpp [hunk 1]
```diff
@@ -612,10 +612,9 @@ void PipelineSolver::populateReadyList(
   }
 
   if (UseCostHeur) {
-    std::sort(ReadyList.begin(), ReadyList.end(),
-              [](std::pair<int, int> A, std::pair<int, int> B) {
-                return A.second < B.second;
-              });
+    llvm::sort(ReadyList, [](std::pair<int, int> A, std::pair<int, int> B) {
```
Comment (MaskRay) [inline llvm/lib/Target/AMDGPU/AMDGPUIGroupLP.cpp:615]: less_second

**Review by MaskRay — APPROVED** [review]

File: llvm/lib/Target/AMDGPU/AMDGPUIGroupLP.cpp (hunk 1 above)
Comment (kazutakahirata) [reply llvm/lib/Target/AMDGPU/AMDGPUIGroupLP.cpp]: Fixed in the latest iteration.  Thanks!

**Comment by RKSimon** [comment]
@kazutakahirata Please can you take a look at the test failures on EXPENSIVE_CHECKS builds?

**Comment by kazutakahirata** [comment]
> @kazutakahirata Please can you take a look at the test failures on EXPENSIVE_CHECKS builds?
> 
> https://lab.llvm.org/buildbot/#/builders/16/builds/500

Will do.

**Comment by kazutakahirata** [comment]
> > @kazutakahirata Please can you take a look at the test failures on EXPENSIVE_CHECKS builds?
> > https://lab.llvm.org/buildbot/#/builders/16/builds/500
> 
> Will do.

I've reverted the whole thing as https://github.com/llvm/llvm-project/commit/fef144cebb378f550ef098d370316554d647f625 for simplicity even though the culprit is under those hunks under `llvm/lib/Target`.

**Pull request merged**